Only create render layers when Opacity is animating.

grokys opened the pull request:
- Add `AvaloniaObject.IsAnimating(property)` to allow us to check if a property has an active animation on a property
- Only create a render layer when a property:
  1. Has an animating `Opacity` property _and_
  2. Has child controls
- Don't render controls with 0 opacity

We should also create a new render layer when a control has an animating `RenderTransform` but that would be best to put in a separate PR I think.

grokys committed:
Added AvaloniaObject.IsAnimating(property).

Added a method to test whether a property on an object is currently animating.

grokys committed:
Display # of layers next to FPS counter.

grokys committed:
Only create render layers when Opacity is animating.

Previously we were creating a new render layer for all controls with `Opacity != 1`. This was causing the `Calendar` page in ControlCatalog to render _really_ slowly. Instead, only create a new render layer when `Opacity` is being animated.

Also don't even render controls where `Opacity == 0`.

grokys committed:
Don't create render layers for non-parent controls.

Only create a new render layer when a control has both an animating opacity _and_ children.

kekekeks commented on src/Avalonia.Base/PriorityValue.cs:
```diff
@@ -52,6 +52,8 @@ internal class PriorityValue
             _validate = validate;
         }
 
+        public bool IsAnimating => ValuePriority < 0 && GetLevel(ValuePriority).ActiveBindingIndex != -1;
```
We probably need more docs on priority value system, since this line has some "magic" numbers and it's hard to understand what it actually checks

kekekeks reviewed (approved):
LGTM

grokys committed:
Tried to make IsAnimating logic clearer.

grokys commented:
There is a failing test here which has made me realize that some of the logic is wrong. Need to fix this before merging.

grokys committed:
Run D2D render tests in ncrunch.

grokys committed:
Simplified DeferredRendererTests,

grokys committed:
Fix layers with opacity.

We were pushing the opacity twice.

grokys committed:
Fix opacity mask rendering.

We're potentially creating a new immutable version of the opacity mask each time the `VisualNode` is updated. Need to implement comparison between mutable and immutable brushes.

grokys committed:
Fix failing skia render test

Make sure we restore the transform in `EndRender` and use the same order of pushing/popping operations that `ImmediateRenderer` uses.
